### ISSUE TITLE
Take photo crash and record issue

### ISSUE THREAD

**Issue by 0k1019** [gorastudio/SCNRecorder#19]
Hello, Thank you for wonderful project.

I bump up the library version to 2.0.0 to 2.2.0.
After I update version, I can't take photo with crash log. And no video after video recording.

I have some devices and I test "SCNRecorder". 
- iPad mini 5(iOS 13.6)
- iPhone 11 Pro Max(iOS 13.5)
- iPhone XS(iOS 14.1)
- iPad Pro 12.1 2020(iPadOS 14.1)

I test by checkout following project tag 2.0.0, 2.1.3 and 2.2.0. 
And run Example Project. 
In `ARSCNView Example`

### Nice working
- iPad mini 5(iOS 13.6)
- iPhone 11 Pro Max(iOS 13.5)
- iPhone XS(iOS 14.1) (NavigationItem bug)

### Not working
- iPad Pro 2020(iPadOS 14.1) (NavigationItem bug)
>` StartVideoRecording` Not working

### Nice working
- iPad Pro 12.1 2020(iPadOS 14.1)
- iPhone XS(iOS 14.1)

### Not working
- iPad mini 5(iOS 13.6)
&
- iPhone 11 Pro Max(iOS 13.5)
>` TakePhoto` Not working with error.

```2020-11-16 12:18:16.303362+0900 Example[529:45566] Metal GPU Frame Capture Enabled
2020-11-16 12:18:16.303563+0900 Example[529:45566] Metal API Validation Enabled
2020-11-16 12:18:18.087331+0900 Example[529:45847] [CAMetalLayerDrawable texture] should not be called after already presenting this drawable. Get a nextDrawable instead.
Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60
2020-11-16 12:18:18.089814+0900 Example[529:45768] Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60
```

>` StartVideoRecording` Not working with black video.

### Nice working
- iPad Pro 12.1 2020(iPadOS 14.1)
- iPhone XS(iOS 14.1)

### Not working
- iPad mini 5(iOS 13.6)
&
- iPhone 11 Pro Max(iOS 13.5)
>` TakePhoto` Not working with error.

```2020-11-16 12:28:26.076176+0900 Example[555:49819] Metal GPU Frame Capture Enabled
2020-11-16 12:28:26.076353+0900 Example[555:49819] Metal API Validation Enabled
Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60
2020-11-16 12:28:27.593305+0900 Example[555:50010] Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60
```
>` StartVideoRecording` Not working with black video.

## Clue
I try to find clue
I find ` TakePhoto`
`VTCreateCGImageFromCVPixelBuffer` function is not working correctly iOS 13.

I can't find any clue ` StartVideoRecording`

I think OS version dependent on this project. 
Please check.

Hope you find reason soon!

**Comment by v-grigoriev** [comment]
Thanks for your findings!
I was able to reproduce the issue and I'm working on fixing it.

**Comment by v-grigoriev** [comment]
Actually I was wrong.
I was able to reproduce the issue using MTLPixelFormat.bgra10_xr_srgb, but it should be an impossible format for any SCNView related views.
@0k1019 Can you please tell me what is the pixelFormat of CAMetalLayer you have in iOS 13 on iPad and iPhone?
You can grab it, for example near `takePhoto` function call.
e.g. `sceneView.recordableLayer!.pixelFormat.rawValue`

**Comment by 0k1019** [comment]
@v-grigoriev 
I run `print(sceneView.recordableLayer!.pixelFormat.rawValue)`.
The result is 81 (MTLPixelFormat.bgra8Unorm_srgb) in iOS 13 on iPhone and iPad.

Below text and photo are about log and Fatal error screen shot.

> 2020-11-23 10:46:09.518798+0900 Example[277:5553] Metal GPU Frame Capture Enabled
2020-11-23 10:46:09.518993+0900 Example[277:5553] Metal API Validation Enabled
before takePhoto function call
sceneView.recordableLayer!.pixelFormat.rawValue:  81
after takePhoto function call
sceneView.recordableLayer!.pixelFormat.rawValue:  81
Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60
2020-11-23 10:46:18.099552+0900 Example[277:5703] Fatal error: Unexpectedly found nil while unwrapping an Optional value: file SCNRecorder/ImageOutput.swift, line 60

<img width="474" alt="스크린샷 2020-11-23 오전 10 46 43" src="https://user-images.githubusercontent.com/11485593/99923466-38abad80-2d79-11eb-8c1e-458196091ecd.png">

**Comment by v-grigoriev** [comment]
@0k1019  Can you please test with 2.3.0?

**Comment by 0k1019** [comment]
@v-grigoriev 
Thank you. It works really well in example project!

**Comment by v-grigoriev** [comment]
Sure, starting with 2.1 to support iOS 14 on iPhone 12 and iPad Pro 2020 I changed the way to capture pixel buffers from MTLTexture. Instead of `getBytes`, I started using `iosurface` to create CVPixelBuffer. And it works on iOS 14 without any problem.
But it seems iOS 13 Video Toolbox doesn't support such pixel buffers, so I reverted the change for iOS 13.

And now I have an idea of how to make all of this even better. I'm working on a solution that uses Metal Performance Shaders to properly and fast capture pixel buffers. And I'm going to add unit tests for testing with all supported pixel formats on various devices and iOS versions to prevent such crashes in the future.
